agents/README.md

```diff
@@ -0,0 +1,51 @@
+# Transporter Agents
+Specialized services for executing file transfers.
+
+## Responsibilities
+Agents are the workhorses within Transporter. A single agent connects to a
+queue resource initialized by the [Manager](../manager/README.md), and:
+* Consumes messages from the queue's "request" topic, using the data to:
+   1. Perform storage-specific validation of the request
+   2. Initialize the upload in the target storage system
+   3. Push an initial progress marker onto the queue's "progress" topic
+* Consumes messages from the queue's "progress" topic, using the data to:
+   1. Transfer a chunk of data from source to target
+   2. Determine if the pushed chunk was the last required chunk
+   3. Push either a new progress marker onto the queue's "progress" topic,
+      or a success message onto the queue's "result" topic, depending on
+      the result of transferring the chunk
+* Converts errors raised during the initialization and step streams into
+  failure messages, and pushes those messages onto the queue's "result" topic
+
+## Implementations
+Common agent code is captured in a [template](template/README.md). So far,
+we've produced the following concrete implementation of the template:
+* [Echo](echo/README.md) (no-op)
+* [AWS->GCP](aws-to-gcp/README.md)
+
+## Running Locally
+To run an agent locally, first run the [Manager](../manager/README.md)
```

Comment:
same with git hash

also maybe make this have a number/bullet point e.g: 

`Step 1. To run an agent locally, first run the [Manager](../manager/README.md)`